strawberryfields/io.py

```diff
@@ -244,6 +245,66 @@ def is_free_param(param):
     return prog
 
 
+def serialize_program(prog, eng=None):
+    """Converts a Strawberry Fields program into valid Strawberry Fields code.
```

Comment:
Maybe this requires a better function name.

I'm not such a fan of `serialize_program`, for several reasons:

* It doesn't really emphasize the end result (i.e., produces a string containing valid Python code)

* It could potentially be confused with `io.save`, which is ideally what we want users to use when they are thinking 'what is the best way to serialize my SF program'.

* While it _is_ a form of serialization, it's not a very *good* form of serialization. Firstsly, you can't really de-serialize it easily (I'm not sure running it through the Python interpreter counts 😆). Secondly, it makes a few assumptions when serializing to 'fill in the gaps'.

Maybe `generate_python_code`, `generate_python`, or some other permutation thereof?

Reply:
Thanks for the comments @josh146. What you're saying makes sense, and I'll update it to something better. :slightly_smiling_face:  :+1: